Objects/codeobject.c

```diff
@@ -2268,3 +2268,76 @@ _PyStaticCode_InternStrings(PyCodeObject *co)
     }
     return 0;
 }
+
+PyCodeObject *
+_Py_MakeShimCode(const _PyShimCodeDef *codedef)
+{
+    PyObject *name = NULL;
+    PyObject *co_code = NULL;
+    PyObject *lines = NULL;
+    PyCodeObject *codeobj = NULL;
+    uint8_t *loc_table = NULL;
+
+    name = _PyUnicode_FromASCII(codedef->cname, strlen(codedef->cname));
+    if (name == NULL) {
+        goto cleanup;
+    }
+    co_code = PyBytes_FromStringAndSize(
+        (const char *)codedef->code, codedef->codelen);
+    if (co_code == NULL) {
+        goto cleanup;
+    }
+    int code_units = codedef->codelen/2;
+    int loc_entries = (code_units + 7)/8;
+    loc_table = PyMem_Malloc(loc_entries);
+    if (loc_table == NULL) {
+        PyErr_NoMemory();
+        return NULL;
+    }
+    for (int i = 0; i < loc_entries-1; i++) {
+         loc_table[i] = 0x80 | (PY_CODE_LOCATION_INFO_NONE << 3) | 7;
+         code_units -= 8;
+    }
+    assert(loc_entries > 0);
+    assert(code_units > 0 && code_units <= 8);
+    loc_table[loc_entries-1] = 0x80 |
+        (PY_CODE_LOCATION_INFO_NONE << 3) | (code_units-1);
+    lines = PyBytes_FromStringAndSize((const char *)loc_table, loc_entries);
+    if (lines == NULL) {
+        goto cleanup;
+    }
+    struct _PyCodeConstructor con = {
+        .filename = &_Py_STR(empty),
```

Comment:
Not a *huge* deal, but I think we typically put *something* here for ease of debugging. Maybe `"<internal>"`?